### ISSUE TITLE
program too large with commit 12d0d

### ISSUE THREAD

**Issue by phlemoine** [SuperHouse/esp-open-rtos#225]
i just rebased on head of the repo (currently at 83689) and found out my program no longer fits the max size due to commit 12d0d. has anyone any recommendation on how to adjust this commit or a local change i could use to overcome this issue and still be able to stay aligned with the latest and greatest version?

**Comment by resetnow** [comment]
Have you tried adding `-Os` to CFLAGS?

**Comment by phlemoine** [comment]
i just dit, with this :
CFLAGS:=$(filter-out -O2,$(CFLAGS))
CFLAGS += -Os

however, the error is still reported:
section `.text' will not fit in region `iram1_0_seg'

**Comment by resetnow** [comment]
Most functions should be placed into flash by default, do you explicitly specify any function attributes in your code to force functions into IRAM?

**Comment by phlemoine** [comment]
Nope. All functions run in flash. What puzzles me is what in this commit
(12d0d) is causing this error. Once reverted, it builds fine

**Comment by resetnow** [comment]
This commit increases data memory usage by a few bytes, maybe that's the reason? Try building any of example apps (with `Os`), and maybe `make clean` in your app to rebuild everything with optimization.

**Comment by resetnow** [comment]
Out of curiosity, did you figure it out?

**Comment by phlemoine** [comment]
No, I didn't. I went back to the code changes on that commit, and can't
really spot what would cause such an error.
I had try to build the examples , but that didn't help me with the issue.
I'll conduct some more experiment next week and will post the results.

>
